CMakeLists.txt

```diff
@@ -33,17 +33,25 @@ set(VERSION_CACHE "v17")
 
 # get software version info from git
 
-find_package(Git)
-
-if(GIT_FOUND)
-  execute_process(
-    COMMAND "${GIT_EXECUTABLE}" describe --abbrev=0 --tags
-    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
-    OUTPUT_VARIABLE VERSION_SOFTWARE
-    OUTPUT_STRIP_TRAILING_WHITESPACE )
-else()
-  message(WARNING "Because Git was not found, the version of the pairinteraction software cannot be determined.")
-  set(VERSION_SOFTWARE "v0.0.0")
+if(NOT VERSION_SOFTWARE)
+  find_package(Git)
+  if(GIT_FOUND)
+    execute_process(
+      COMMAND "${GIT_EXECUTABLE}" describe --abbrev=0 --tags
+      WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+      RESULT_VARIABLE GIT_DESCRIBE_RESULT
+      OUTPUT_VARIABLE GIT_DESCRIBE_OUTPUT
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if(GIT_DESCRIBE_RESULT EQUAL 0)
+      set(VERSION_SOFTWARE "${GIT_DESCRIBE_OUTPUT}" CACHE STRING "Set software version" FORCE)
+    else()
+      message(WARNING "Git failed to determine the version of the pairinteraction software.")
+      set(VERSION_SOFTWARE "v0.0.0" CACHE STRING "Set software version" FORCE)
+    endif()
+  else()
+    message(WARNING "Because Git was not found, the version of the pairinteraction software cannot be determined.")
+    set(VERSION_SOFTWARE "v0.0.0" CACHE STRING "Set software version" FORCE)
```

Comment:
We could also think about
```cmake
set(VERSION_SOFTWARE "${VERSION_SOFTWARE_DEFAULT}" CACHE STRING "Set software version" FORCE)
```
and further above do
```cmake
set(VERSION_SOFTWARE_DEFAULT "v0.9.8-alpha")
```
which has to be bumped manually with every release. This would always guarantee a more sensible version number but at the expense of some manual work at release time. I think we should also write a release checklist at some point.